openfeature-provider/go/confidence/swap_wasm_resolver_api_test.go

```diff
@@ -338,3 +472,137 @@ func TestErrInstanceClosed(t *testing.T) {
+
+	stickyRequest := createResolveWithStickyRequest(
+		createTutorialFeatureRequest(),
+		nil,   // empty materializations
+		true,  // failFast
+		false, // notProcessSticky
+	)
+
+	response, err := wasmResolver.ResolveWithSticky(stickyRequest)
+	if err != nil {
+		t.Fatalf("Unexpected error resolving tutorial-feature flag with sticky: %v", err)
+	}
+
+	if response == nil {
+		t.Fatal("Expected non-nil response")
+	}
+
+	successResult, ok := response.ResolveResult.(*resolver.ResolveWithStickyResponse_Success_)
+	if !ok {
+		t.Fatal("Expected success result from ResolveWithSticky")
+	}
+
+	resolveResponse := successResult.Success.Response
+	if len(resolveResponse.ResolvedFlags) != 1 {
+		t.Fatalf("Expected 1 resolved flag, got %d", len(resolveResponse.ResolvedFlags))
+	}
+
+	resolvedFlag := resolveResponse.ResolvedFlags[0]
+
+	if resolvedFlag.Flag != "flags/tutorial-feature" {
+		t.Errorf("Expected flag 'flags/tutorial-feature', got '%s'", resolvedFlag.Flag)
+	}
+
+	expectedVariant := "flags/tutorial-feature/variants/exciting-welcome"
+	if resolvedFlag.Variant != expectedVariant {
+		t.Errorf("Expected variant '%s', got '%s'", expectedVariant, resolvedFlag.Variant)
+	}
+
+	if resolvedFlag.Reason.String() != "RESOLVE_REASON_MATCH" {
+		t.Errorf("Expected reason RESOLVE_REASON_MATCH, got %v", resolvedFlag.Reason)
+	}
+
+	if resolvedFlag.Value == nil {
+		t.Fatal("Expected non-nil value in resolved flag")
+	}
+
+	fields := resolvedFlag.Value.GetFields()
+	if fields == nil {
+		t.Fatal("Expected fields in resolved value")
+	}
+
+	expectedMessage := "We are very excited to welcome you to Confidence! This is a message from the tutorial flag."
+	messageValue, hasMessage := fields["message"]
+	if !hasMessage {
+		t.Error("Expected 'message' field in resolved value")
+	} else if messageValue.GetStringValue() != expectedMessage {
+		t.Errorf("Expected message '%s', got '%s'", expectedMessage, messageValue.GetStringValue())
+	}
+
+	expectedTitle := "Welcome to Confidence!"
+	titleValue, hasTitle := fields["title"]
+	if !hasTitle {
+		t.Error("Expected 'title' field in resolved value")
+	} else if titleValue.GetStringValue() != expectedTitle {
+		t.Errorf("Expected title '%s', got '%s'", expectedTitle, titleValue.GetStringValue())
+	}
+}
+
+func TestSwapWasmResolverApi_ResolveFlagWithStickyRules_MissingMaterializations(t *testing.T) {
+	ctx := context.Background()
+	runtime := wazero.NewRuntimeWithConfig(ctx, wazero.NewRuntimeConfig())
+	defer runtime.Close(ctx)
+
+	flagLogger := NewNoOpWasmFlagLogger()
+	stickyState := createStateWithStickyFlag()
+	accountId := "test-account"
+
+	swap, err := NewSwapWasmResolverApi(ctx, runtime, defaultWasmBytes, flagLogger, stickyState, accountId)
+	if err != nil {
+		t.Fatalf("Failed to create SwapWasmResolverApi: %v", err)
+	}
+	defer swap.Close(ctx)
+
+	stickyRequest := createResolveWithStickyRequest(
+		&resolver.ResolveFlagsRequest{
+			Flags:        []string{"flags/sticky-test-flag"},
+			Apply:        false,
```

Comment:
this is hard coded to true in our implementation so let's keep it like that?
```suggestion
			Apply:        true,
```